File: ivan/ql-project/src/net/iplantevin/ql/ast/visitors/IStatementVisitor.java
package net.iplantevin.ql.ast.visitors;

import net.iplantevin.ql.ast.statements.Block;
import net.iplantevin.ql.ast.statements.Computation;
import net.iplantevin.ql.ast.statements.If;
import net.iplantevin.ql.ast.statements.IfElse;
import net.iplantevin.ql.ast.statements.Question;

/**
 * @author Ivan
 */
public interface IStatementVisitor<T> {
    public T visit(Block block);

    public T visit(Computation computation);

    public T visit(If ifStat);

    public T visit(IfElse ifElse);

    public T visit(Question question);
}

File: rperz/QSLib/QSLib/Visitors/IStatementVisitor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using QSLib.AST.Statements;
using QSLib.AST;

namespace QSLib.Visitors
{
    public interface IStatementVisitor<T>
    {
        T Visit(Form form);
        T Visit(IfStatement ifStatement);
        T Visit(Question question);
        T Visit(ComputedQuestion computedQuestion);
        T Visit(CodeBlock codeBlock);
    }
}
